tests/e2e/util.go

```diff
@@ -6823,3 +6826,90 @@ func getVmdkPathFromVolumeHandle(sshClientConfig *ssh.ClientConfig, masterIp str
 	vmdkPath := result.Stdout
 	return vmdkPath
 }
+
+func checkVcServicesHealth(ctx context.Context, host string, timeout ...time.Duration) {
+	var pollTime time.Duration
+	if len(timeout) == 0 {
+		pollTime = pollTimeout * 6
+	} else {
+		pollTime = timeout[0]
+	}
+	//list of default stopped services in VC
+	var defaultStoppedServicesList = []string{"vmcam", "vmonapi", "vmware-imagebuilder", "vmware-netdumper",
+		"vmware-rbd-watchdog", "vmware-vcha"}
+	waitErr := wait.PollUntilContextTimeout(ctx, poll, pollTime, true,
+		func(ctx context.Context) (bool, error) {
+			var pendingServiceslist []string
+			var noAdditionalServiceStopped = false
+			sshCmd := fmt.Sprintf("service-control --%s", statusOperation)
+			framework.Logf("Invoking command %v on vCenter host %v", sshCmd, host)
+			result, err := fssh.SSH(ctx, sshCmd, host, framework.TestContext.Provider)
+			if err != nil || result.Code != 0 {
+				fssh.LogResult(result)
+				return false, fmt.Errorf("couldn't execute command: %s on vCenter host: %v", sshCmd, err)
+			}
+			framework.Logf("Command %v output is %v", sshCmd, result.Stdout)
+			// getting list of services which are stopped currently
+			services := strings.SplitAfter(result.Stdout, svcRunningMessage+":")
+			servicesStoppedInVc := strings.Split(services[1], svcStoppedMessage+":")
+			// getting list if pending services if any
+			if strings.Contains(servicesStoppedInVc[0], "StartPending:") {
+				pendingServiceslist = strings.Split(servicesStoppedInVc[0], "StartPending:")
+				pendingServiceslist = strings.Split(strings.Trim(pendingServiceslist[1], "\n "), " ")
```

Comment:
This might throw error if there are no pending services to be started in VC. Check if  length oflist Pendingservices is greater than 0.

Reply:
If no pending services there, it won't come inside this IF loop. Also this Key "**StartPending**:" won't be there.